mimiron/core/commands/deploy.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from . import Command as _Command
+from .. import io
+from .. import constants as const
+from ..util.time import pretty_print_datetime
+from ...vendor.git_extensions import extensions as git_ext
+
+
+class Deploy(_Command):
+    MAX_SUMMARY_LIMIT = 50
+
+    def _validate_and_configure(self):
+        super(self.__class__, self)._validate_and_configure()
+        self.deployment_repo = self.config['TF_DEPLOYMENT_REPO']
+
+        self.show_last_limit = int(self.kwargs['show_last_limit'])
+        self.should_push = self.kwargs['should_push']
+
+    def _prompt_commit_selection(self):
+        io.info('displaying @~%s most recent commits' % self.show_last_limit)
+        table_data = [
+            ['id', 'commit id', 'message', 'author', 'committed at'],
+        ]
+        limit = self.__class__.MAX_SUMMARY_LIMIT
```

Comment:
Good question.

1. First, `MAX_SUMMARY_LIMIT` isn't private. A getter/setter would make sense if I needed to expose a private instance variable but I don't need that here.
1. `MAX_SUMMARY_LIMIT` is a public constant and by general convention, if you tinker with a constant, you're going to have a bad time (this is why I was OK with having it public, otherwise I would have made it private and pass it during instantiation).
1. This is more just related data I've attached to my class. Don't think of this as an instance variable. I've placed it here because I think it's more convenient than storing it inside a `constants.py` file.

To (3), you'll often see this kind of thing in other Python projects. Run `help(class)` inside your repl. For example:

```
>>> class X(object):
...     DATA_X_1 = "hello"
...
>>> help(X)
```

```
class X(__builtin__.object)
 |  Data descriptors defined here:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  DATA_X_1 = 'hello'
```

The other thing is I like doing `self.__class__.DATA_ATTR` because it makes it easier to refactor the class if I rename it.